perf_test/sparse/KokkosSparse_sptrsv_aux.hpp

```diff
@@ -0,0 +1,626 @@
+  cusparseOperation_t transL = (col_majorL ? CUSPARSE_OPERATION_TRANSPOSE : CUSPARSE_OPERATION_NON_TRANSPOSE);
+  if (std::is_same<scalar_t, double>::value == true) {
+    cusparseDcsrsv2_bufferSize (handle, transL, nrows, nnzL, descrL,
+                                reinterpret_cast <double*> (valuesL.data()), row_mapL.data(), entriesL.data(), infoL,
+                                &pBufferSize);
+  } else {
+    cusparseZcsrsv2_bufferSize (handle, transL, nrows, nnzL, descrL,
+                                reinterpret_cast <cuDoubleComplex*> (valuesL.data()), row_mapL.data(), entriesL.data(), infoL,
+                                &pBufferSize);
+  }
+  cudaMalloc((void**)&pBufferL, pBufferSize);
+
+  // ==============================================
+  // step 3: analysis
+  std::cout << "  Lower-Triangular" << std::endl;
+  timer.reset ();
+  const cusparseSolvePolicy_t policy = CUSPARSE_SOLVE_POLICY_USE_LEVEL;
+  if (std::is_same<scalar_t, double>::value == true) {
+    status = cusparseDcsrsv2_analysis (handle, transL, nrows, nnzL, descrL,
+                                       reinterpret_cast <double*> (valuesL.data()), row_mapL.data(), entriesL.data(),
+                                       infoL, policy, pBufferL);
+  } else {
+    status = cusparseZcsrsv2_analysis (handle, transL, nrows, nnzL, descrL,
+                                       reinterpret_cast <cuDoubleComplex*> (valuesL.data()), row_mapL.data(), entriesL.data(),
+                                       infoL, policy, pBufferL);
+  }
+  std::cout << "  Cusparse Symbolic Time: " << timer.seconds() << std::endl;
+  if (CUSPARSE_STATUS_SUCCESS != status) {
+    std::cout << "analysis status error name " << (status) << std::endl;
+  }
+  // L has unit diagonal, so no structural zero is reported.
+
+  int structural_zero;
+  status = cusparseXcsrsv2_zeroPivot (handle, infoL, &structural_zero);
+  if (CUSPARSE_STATUS_ZERO_PIVOT == status){
+    printf("L(%d,%d) is missing\n", structural_zero, structural_zero);
+  }
+
+  // ==============================================
+  // Preaparing for the first solve
+  //> create the known solution and set to all 1's ** on host **
+  host_scalar_view_t sol_host ("sol_host", nrows);
+  Kokkos::deep_copy (sol_host, ONE);
+
+  // > create the rhs ** on host **
+  // A*sol generates rhs: rhs is dense, use spmv
+  host_scalar_view_t rhs_host ("rhs_host", nrows);
+  KokkosSparse::spmv ("N", ONE, Mtx, sol_host, ZERO, rhs_host);
+
+  // ==============================================
+  // step 1: apply forward-pivot to rhs on the host
+  host_scalar_view_t tmp_host ("temp", nrows);
+  forwardP_supernode<scalar_t> (nrows, perm_r, 1, rhs_host.data(), nrows, tmp_host.data(), nrows);
+
+  // copy rhs to the default host/device
+  scalar_view_t rhs ("rhs", nrows);
+  scalar_view_t sol ("sol", nrows);
+  Kokkos::deep_copy (rhs, tmp_host);
+
+  // ==============================================
+  // step 2: solve L*y = x
+  Kokkos::fence ();
```

Comment:
Note that this only fences Kokkos' default CUDA stream.  Could that possibly be different than cuSPARSE's stream (that comes in through the handle)?